ACPs/77-reinventing-subnets/README.md

```diff
@@ -103,22 +113,34 @@ After the `RegisterSubnetValidatorTx` is accepted on the P-Chain, the Subnet Val
 
 When any validator is removed from the set (whether forcefully or per the validator's request), the P-Chain will also send a warp message to the Subnet notifying it of the validator set removal. It is up to the Subnet on how to handle such a message, especially if unexpected. A validator's stake could continue to remain locked for an extended period of time after this point, for example.
 
-Note: There is no `EndTime` specified in this transaction. Subnet Validators are only evicted when an `EvictSubnetValidatorTx` or `ExitValidatorTx` is issued.
+This transaction is, by design, not required to be submitted by the validator themselves. With the Ed25519 signature, the validator guarantees that they can only be added to the validator set if `Signer` corresponds to the `blsPublicKey` and `Balance` >= `balance`. The `RegisterSubnetValidatorTx` is considered invalid if those two properties are not satisfied.
+
+When the `RegisterSubnetValidatorTx` transaction is issued, `Balance` (but not `balance`) is required to cover two weeks worth of the continuous fee based on the market price at time of registry. This matches the Primary Network's `MinStakeDuration`. Subnets are not required to have a `MinStakeDuration` of two weeks and can remove a validator at any time.
+
+Note: There is no `EndTime` specified in this transaction. Subnet Validators are only removed when a `SetSubnetValidatorWeightTx` sets a validator's weight to `0` or `ExitValidatorTx` is issued.
+
+#### SetSubnetValidatorWeightTx
 
-#### EvictSubnetValidatorTx
+`SetSubnetValidatorWeightTx` is used to modify the voting weight of a Subnet Validator. For this transaction to be valid, 67% of the current Subnet Validator weight must sign the warp message included in `Message`. The fee to pay for this transaction is taken from the `Balance` of the Subnet Validator being modified. This enables Subnet Validators to automatically broadcast this transaction to the P-Chain once the multisig has sufficient participation. Applications of this transaction could include:
```

Comment:
I'm concerned that the balance of the validator is used for this.

Reply:
Yeah I think the incentive is on the validator/delegator who is modifying the weight to send this tx to the P-Chain. Modified to BaseTx.